consoles/video_stream.pm

```diff
@@ -120,15 +140,35 @@ sub connect_remote_video ($self, $url) {
         $self->{dv_timings} = _v4l2_ctl($url, '--get-dv-timings');
     }
 
-    my $cmd = $self->_get_ffmpeg_cmd($url);
-    my $ffmpeg;
-    $self->{ffmpegpid} = open($ffmpeg, '-|', @$cmd)
-      or die "Failed to start ffmpeg for video stream at $url";
-    # make the pipe size large enough to hold full frame and a bit
-    my $frame_size = $bmwqemu::vars{VIDEO_STREAM_PIPE_BUFFER_SIZE} // DEFAULT_VIDEO_STREAM_PIPE_BUFFER_SIZE;
-    fcntl($ffmpeg, Fcntl::F_SETPIPE_SZ, $frame_size);
-    $self->{ffmpeg} = $ffmpeg;
-    $ffmpeg->blocking(0);
+    if ($url =~ m^ustreamer://^) {
+        my $dev = ($url =~ m^ustreamer://(.*)^)[0];
+        my $sink_name = "raw-sink$dev";
+        $sink_name =~ s^/^-^g;
+        my $cmd = $self->_get_ustreamer_cmd($dev, $sink_name);
+        my $ffmpeg;
+        $self->{ustreamerpid} = open($ffmpeg, '-|', @$cmd)
+          or die "Failed to start ustreamer for video stream at $url";
+        $self->{ustreamer_pipe} = $ffmpeg;
+        my $timeout = 100;
+        while ($timeout and !-f "/dev/shm/$sink_name") {
+            sleep(0.1);
+            $timeout -= 1;
+        }
+        die "ustreamer startup timeout" if !$timeout;
```

Comment:
```suggestion
        die "ustreamer startup timeout" if $timeout < 0;
```

Reply:
But the loop exits on $timeout == 0, so if anything then <=